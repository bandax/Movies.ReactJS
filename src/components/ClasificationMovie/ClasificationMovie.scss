ul.movies-types {    
    list-style-type: none;
    margin: 0;
    padding: 0;
    overflow: hidden;    
  
  
  li.movie-type {
    float: left;
  }
  
  li.movie-type a {
    display: block;
    color: white;
    text-align: center;
    padding: 16px;
    text-decoration: none;
  }
  
  li.movie-type a:hover {
    background-color: #111111;
  }

}
