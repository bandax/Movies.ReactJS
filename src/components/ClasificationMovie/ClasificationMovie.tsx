import * as React from 'react';
import './ClasificationMovie.scss';
import { IClasification } from '../../interfaces/IClasificationMovie';

interface IClasificationMovieProps {
    clasificationTypes: IClasification[];
}

const ClasificationMovie: React.FunctionComponent<IClasificationMovieProps>
    = (props: IClasificationMovieProps) => {
        const movieTypes = props.clasificationTypes;

        return (
            <ul className="movies-types">
                {
                    movieTypes.map(({ id, name }) => {
                        return (
                            <li key={id} className="movie-type">
                                <a href="">{name}</a>
                            </li>
                        );
                    })
                }
            </ul>
        );
    }

export { ClasificationMovie };
