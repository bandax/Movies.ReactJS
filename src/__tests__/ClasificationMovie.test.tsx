import * as React from 'react';
import { ClasificationMovie } from '../components/ClasificationMovie/ClasificationMovie';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import clasificationTypes from '../data/clasifications.json';

describe('<Footer />', () => {
  it('render a Footer Component', async () => {
    const { asFragment } = render(
      <ClasificationMovie clasificationTypes={clasificationTypes} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
