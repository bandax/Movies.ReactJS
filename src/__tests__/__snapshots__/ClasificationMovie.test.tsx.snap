// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> render a Footer Component 1`] = `
<DocumentFragment>
  <ul
    class="movies-types"
  >
    <li
      class="movie-type"
    >
      <a
        href=""
      >
        Documentary
      </a>
    </li>
    <li
      class="movie-type"
    >
      <a
        href=""
      >
        Comedy
      </a>
    </li>
    <li
      class="movie-type"
    >
      <a
        href=""
      >
        Horror
      </a>
    </li>
    <li
      class="movie-type"
    >
      <a
        href=""
      >
        Action & Adventure
      </a>
    </li>
    <li
      class="movie-type"
    >
      <a
        href=""
      >
        Drama
      </a>
    </li>
    <li
      class="movie-type"
    >
      <a
        href=""
      >
        Romance
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
